clients/spark-extensions-base/src/test/java/org/projectnessie/spark/extensions/AbstractSparkSqlTest.java

```diff
@@ -476,37 +480,62 @@ void useShowReferences() throws NessieNotFoundException {
     assertThat(sql("SHOW REFERENCE IN nessie")).containsExactly(row("Branch", refName, hash));
   }
 
-  @Test
-  void mergeReferencesIntoMain() throws NessieConflictException, NessieNotFoundException {
-    List<Object[]> commits =
-        commitAndReturnLog(refName).stream()
-            .map(AbstractSparkSqlTest::withoutHashAndTime)
-            .collect(Collectors.toList());
+  private Object[] relevantFromMerge(Object[] a) {
+    return new Object[] {a[2], removeInternalProperties((Map<String, String>) a[5])};
+  }
 
-    sql("MERGE BRANCH %s INTO main IN nessie", refName);
-    // here we are skipping commit time as its variable
+  private static Map<String, String> removeInternalProperties(Map<String, String> properties) {
+    return properties.entrySet().stream()
+        .filter(e -> !e.getKey().startsWith("_"))
+        .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+  }
 
-    assertThat(
-            sql("SHOW LOG main IN nessie").stream()
-                .map(AbstractSparkSqlTest::sqlResultWithoutHashAndTime)
-                .collect(Collectors.toList()))
-        .containsExactlyElementsOf(commits);
+  private Object[] mergedCommits(Object[] a, Object[] b) {
+    @SuppressWarnings("unchecked")
+    Map<String, String> mapA = (Map<String, String>) a[5];
+    @SuppressWarnings("unchecked")
+    Map<String, String> mapB = (Map<String, String>) b[5];
```

Comment:
can we at add a comment here what these magic numbers mean?
or improve names like `commitPropsB` or the like?
or just move this method next to the one that returns the `Object[]` originally? (so its easier to understand what these array indices mean...suboptimal api)

Reply:
See added `SparkCommitLogEntry`